Compare View

dubachti opened the pull request:
Adds the functionality to mark courses and compare them in a table and radar graph.
- Added `CompareView.vue` for comparing courses.
- Updated `CourseView.vue`to show a 'compare' button.
- Added store so selected courses survive reloads.

Compare button:
<img width="1207" height="179" alt="image" src="https://github.com/user-attachments/assets/08e08288-8ce0-423c-a20d-caa36f5c45de" />

Course comparison page:
<img width="1212" height="984" alt="image" src="https://github.com/user-attachments/assets/852cd17e-0a2b-47e8-a352-17f3ed6f237a" />

Sort and see distribution:
<img width="839" height="459" alt="image" src="https://github.com/user-attachments/assets/2d59c8b8-5f1e-465e-91c5-2b2f6e8a441c" />

I'm not super experienced in web dev, let me know if anything needs some improvement. :)

4ndriH commented:
Could you make the course numbers or names in the table open the course page in a new tab when clicked?
otherise you have to search for it again when you want to look at the reviews

EvilBabyDemon commented:
Change in 
src/App.vue 
Line 153:
```
<div v-if="$vuetify.display.smAndUp">
```
to this
```
<div v-if="$vuetify.display.mdAndUp">
```
With the added icon there is too much stuff if the window is a specific size width (600-700ish px)
(for some reason github decided that code suggestions are not allowed to happen too far away from edited code...

EvilBabyDemon commented on src/views/CompareView.vue:
```diff
@@ -0,0 +1,211 @@
+          >
+            <template v-slot:item.courseNumber="{ item }">
+              <div>
+                <div class="font-weight-bold">{{ item.courseNumber }}</div>
+                <div class="text-caption text-medium-emphasis">{{ item.courseName }}</div>
```
Makes the course number small and the course name big. I assume for most people the name is the more important one.

dubachti commented:
Thanks for the review. I accepted all your suggestions and  fixed the following:
- course names overlapping chart: 
   - removed course id from legend
   - move legend below chart if `display.smAndDown`
   - slightly more trimming of course name
 - open course in new tab: made the chips and course names in the table clickable

Let me know if I missed something.
